src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -68,14 +56,8 @@ public ProcessMetrics(ProcessInstrumentationOptions options)
             "process.memory.virtual",
             () =>
             {
-                if (!this.virtualMemoryUsageBytes.HasValue)
-                {
-                    this.Snapshot();
-                }
-
-                var value = this.virtualMemoryUsageBytes.Value;
-                this.virtualMemoryUsageBytes = null;
-                return value;
+                this.currentProcess.Refresh();
```

Comment:
Do we really need to keep using the same instance and do `Refresh` each time, or just creating a new Process instance is sufficient? Would the `Refresh` call here on the same instance cause race condition?

Reply:
Race condition is possible here. I don't know what's the guarantee of the .Refresh() API so if we want to do Refresh on each callback, locking it might be the safest.

Not sure which one is cheaper, refresh + lock or creating a new instance each time. I'll do some experiments.

Reply:
I've benchmarked the above 2 options https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/717#issue-1415674535.
There is no significance difference in time. I updated the code to use the latter option.

Reply:
Thanks!